### DISCUSSION TITLE
Spherical particles within a medium: modelling heat flows

### DISCUSSION THREAD

**Discussion by WilkAndy** [idaholab/moose#23262] [Q&A Modules: General] [answered]
Hi everyone,

A collaborator of mine is considering the problem in which a hot fluid flows through a medium that contains a number of cool, solid spheres.  Let's call the space in which the fluid flows the "interstellar space" and the spheres "planets".  The fluid slowly heats the planets, via heat transfer and heat conduction through the spherical body, which cools the fluid via heat transfer.   Let's think of ways this could be modelled.

Here's one way: create a conforming mesh (so the mesh contains the interstellar space (eg block=1) and the planets (eg block=2)).  Use NavierStokes or PorousFlow to flow through the interstellar space, use heat conduction to solve heat conduction in the planets, and a heat-transfer coefficient between them both.

But my collaborator is wondering: _what if the planets are spherical and small,_ so that perhaps their temperature distribution can be considered a function of radius alone?  (If they were big then the parts facing the fluid source would heat first, but in this Discussion, let's assume this is unimportant.)  Then perhaps we could treat them in a separate MultiApp, with a spherically-symmetric type of Problem?   Ideas?

**Comment by GiudGiud** [comment, marked as answer]
Hi Andy

Both these approaches are feasible.
In fact something similar is performed using the NEAMS tools for pebble bed reactors. 

The first idea seems feasible though maybe a tad computationally expensive, depending on the fluid flow regime and the size of the meshes for heat conduction and for fluid flow. 

The second idea is performed in this example on the Virtual Test Bed for pebble bed reactor simulations
You need Pronghorn to run these inputs, but taking a look it's fairly obvious what is being done
https://github.com/idaholab/virtual_test_bed/blob/devel/pbfhr/steady/ss1_combined.i
The fluid flow is the ss1 input, then the ss3 input is a coarse mesh representing the pebble locations, and ss4 inputs are pebbles (planets here) that exchange surface boundary conditions. 

The MultiApp are 1D spherical calculations. They receive the surface fluid temperature for the heat conduction problem.
They do not send back any info to the fluid solve. They would be sending a heat flux, but we do that separately here. 

Guillaume

**Reply by WilkAndy** [reply]
Wow, this is great, thank you @GiudGiud .  My collaborator (Julia) will be studying that input file in lots of detail!

**Reply by JuliaHLee** [reply]
Thanks a lot, @WilkAndy  and @GiudGiud!

I am the collaborator who asked for help. It is great to know that there is an example for this kind of problem. I would like to try to run this input files using MultiApp. But when I tried to get information about Pronghorn on the MOOSE website, I found that the access seems to be protected. I assume that I will need to install Pronghorn first to run these input files, right? Could you let me know if I can get access to the information about Pronghorn and installation of the app?

I hope the discussion is still opened to ask further on this issue. It would be great if I get a clue about using the app.

Best regards,
Julia

**Reply by GiudGiud** [reply]
Hello

Pronghorn is export controlled. You ll have to ask for access here and let us know
It can take a couple weeks depending on when you apply and when they meet to consider your application
https://ncrcaims.inl.gov/Identity/Account/Login

@aprilnovak 's thesis has all the equations that are implemented in Pronghorn wrt to porous media with "pebbles", including the distributed spherical simulations
Pronghorn manual can help too
https://inldigitallibrary.inl.gov/sites/sti/sti/Sort_24425.pdf

Guillaume

**Reply by aprilnovak** [reply]
Hi all, thanks for the interest in this! I want to note that the heterogeneous heat conduction method in Pronghorn (i.e. the multiscale method used to approximate particle temperatures distributed through a medium) is almost entirely accessible through open-source parts of MOOSE. I believe there is just one object in Pronghorn (a `LinearCombinationMatchedValueBC` that sets a dirichlet value as `coupled Variable + Postprocessor`) that you'd need to do this open-source. 

As part of other work at ANL, I was planning in the next few months to move that one object into the open-source part of MOOSE (because it's so generic it has nothing to do with nuclear physics), effectively making the heterogeneous heat conduction approach open-source.
